mixer/adapter/kubernetesenv/kubernetesenv.go

```diff
@@ -117,6 +118,11 @@ func (b *builder) Validate() (ce *adapter.ConfigErrors) {
 	return
 }
 
+func init() {
+	// Set up Kubernetes client log to stderr, rather than to file
+	klog.InitFlags(nil)
```

Comment:
Interesting. Why is this needed here but not in Pilot or Galley?

Reply:
That is a good question. The error I was getting was that it fails to write to `/tmp/somefile.log` because it has no permissions (read only). So my initial guess is pilot/galley have write access to tmp, but mixer doesn't because it is `from scratch`

Reply:
But we don't want to write to container FS in general. So I think this needs to be applied generally, not just specially in mixer. Perhaps, look into how istio/log could configure klog in all components?

Reply:
Just double checked. Galley prints to stderr. Pilot doesn't use this outside of tests.

Why Galley prints to stderr without this I am not sure, which is a bit concerning, but I checked the tmp folders of galley and pilot and saw nothing, and checked logs of galley and saw klog stuff and no klog stuff from pilot.

Reply:
I think this may have to do with in-progress migration to klog. I think we still need to make sure that either `klog.InitFlag` is invoked in all the binaries, or that it is configured not to write files.

We had to do quite a bit of hackery around its predecessor glog here https://github.com/istio/glog/blob/master/glog.go, that we also need for klog I think.